src/batch/src/executor/log_row_seq_scan.rs

```diff
@@ -0,0 +1,251 @@
+            metrics
+                .executor_metrics()
+                .row_seq_scan_next_duration
+                .with_guarded_label_values(&metrics.executor_labels(&identity))
+        });
+        // Range Scan
+        // WARN: DO NOT use `select` to execute range scans concurrently
+        //       it can consume too much memory if there're too many ranges.
+        let stream = Self::execute_range(
+            table.clone(),
+            old_epoch.clone(),
+            new_epoch.clone(),
+            chunk_size,
+            histogram.clone(),
+            Arc::new(schema.clone()),
+        );
+        #[for_await]
+        for chunk in stream {
+            let chunk = chunk?;
+            yield chunk;
+        }
+    }
+
+    #[try_stream(ok = DataChunk, error = BatchError)]
+    async fn execute_range(
+        table: Arc<StorageTable<S>>,
+        old_epoch: BatchQueryEpoch,
+        new_epoch: BatchQueryEpoch,
+        chunk_size: usize,
+        histogram: Option<impl Deref<Target = Histogram>>,
+        schema: Arc<Schema>,
+    ) {
+        let pk_prefix = OwnedRow::default();
+
+        let order_type = table.pk_serializer().get_order_types()[pk_prefix.len()];
+        // Range Scan.
+        let iter = table
+            .batch_iter_log_with_pk_bounds(
+                old_epoch.into(),
+                new_epoch.into(),
+                &pk_prefix,
+                (
+                    if order_type.nulls_are_first() {
+                        // `NULL`s are at the start bound side, we should exclude them to meet SQL semantics.
+                        Bound::Excluded(OwnedRow::new(vec![None]))
+                    } else {
+                        // Both start and end are unbounded, so we need to select all rows.
+                        Bound::Unbounded
+                    },
+                    if order_type.nulls_are_last() {
+                        // `NULL`s are at the end bound side, we should exclude them to meet SQL semantics.
+                        Bound::Excluded(OwnedRow::new(vec![None]))
+                    } else {
+                        // Both start and end are unbounded, so we need to select all rows.
+                        Bound::Unbounded
+                    },
+                ),
+            )
+            .await?;
+
+        pin_mut!(iter);
+        loop {
+            let timer = histogram.as_ref().map(|histogram| histogram.start_timer());
+
+            let mut iter = iter.as_mut().map(|r| match r {
+                Ok((op, value)) => {
+                    let (k, row) = value.into_owned_row_key();
+                    // Todo! To avoid create a full row.
+                    let full_row = row
+                        .into_iter()
+                        .chain(vec![Some(ScalarImpl::Int16(op.to_i16()))])
+                        .collect_vec();
+                    let row = OwnedRow::new(full_row);
+                    Ok(KeyedRow::<Bytes>::new(k, row))
```

Comment:
Use `Ok(KeyedRow::<_>::new(k, row))`, and then you don't need to add `use bytes::Bytes`, and then no need to change the `Cargo.toml` to require extra approval.